### PR TITLE
Jwt phase2b

### PR DESCRIPTION
**Pull request by NavinColumbia**
## Description

Jwt auth has been added, endpoints for auth,share permission, schema/models for same.

## Checklist

- [yes ] Tests pass
- [yes] No checkstyle errors
- [yes ] No PMD errors
- [yes ] No compilation/build errors

### CODE REVIEW

File: src/main/java/com/bytecoders/pharmaid/repository/model/UserType.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package com.bytecoders.pharmaid.repository.model;
+
+/** Enum representing the different types of users in the system. */
+public enum UserType {
```
Comment (AlexLoyko) [inline src/main/java/com/bytecoders/pharmaid/repository/model/UserType.java:4]: nice!

File: src/main/java/com/bytecoders/pharmaid/service/SharingPermissionService.java [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+   *
+   * @param ownerId The ID of the user who owns the records.
+   * @param request The sharing request details.
+   * @return String The ID of the created sharing permission.
+   * @throws RuntimeException if either the owner or requester is not found.
+   */
+  public String createSharingRequest(String ownerId, SharingRequest request) {
+
+    User owner =
+        userRepository.findById(ownerId).orElseThrow(() -> new RuntimeException("Owner not found"));
```
Comment (AlexLoyko) [inline src/main/java/com/bytecoders/pharmaid/service/SharingPermissionService.java:42]: you need to throw a specific exception. This should throw a 404

File: src/main/java/com/bytecoders/pharmaid/service/AuthorizationService.java [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+  public boolean canModifyUserRecords(String currentUserId, String targetUserId) {
+
+    if (currentUserId.equals(targetUserId)) {
+      return true;
+    }
+
+    User currentUser =
+        userRepository
+            .findById(currentUserId)
+            .orElseThrow(() -> new RuntimeException("Current user not found"));
```
Comment (AlexLoyko) [inline src/main/java/com/bytecoders/pharmaid/service/AuthorizationService.java:82]: I don't think we should throw in such methods. The effect is that the user CANT modify the records, therefore just return false and throw FORBIDDEN like you already do. applies to other places too

**Comment by NavinColumbia** [comment]
opened https://github.com/NavinColumbia/COMS4156-TeamProject-ByteCoders/pull/23 instead

**Pull request closed without merging**